### PR TITLE
Master relational model aab

### PR DESCRIPTION
**Pull request by aab-odoo**

### CODE REVIEW

File: addons/web/static/src/views/list/list_renderer.js [hunk 1]
```diff
@@ -429,15 +429,30 @@ export class ListRenderer extends Component {
...
-
-    isLocked(record) {
-        return this.props.list._lockedRecord?.id === record.id;
+        if (record.isNew) {
+            return false;
+        }
+        if (this.props.activeActions?.edit === false) {
+            return true;
+        }
+        if (record.isInEdition && !this.isInlineEditable(record) && !record.model.multiEdit) {
```
Comment (aab-odoo) [inline addons/web/static/src/views/list/list_renderer.js:450]: check kanban

File: addons/web/static/src/views/relational_model/static_list.js [hunk 2]
```diff
@@ -210,13 +216,130 @@ export class StaticList extends DataPoint {
...
+    /**
+     * This method is meant to be used in a very specific usecase: when an x2many record is viewed
+     * or edited through a form view dialog (e.g. x2many kanban or non editable list). In this case,
+     * the form typically contains different fields than the kanban or list, so we need to "extend"
+     * the fields and activeFields. If the record opened in a form view dialog already exists, we
+     * modify it's config to add the new fields. If it is a new record, we create it with the
+     * extended config.
+     *
+     * @param {Object} params
+     * @param {Object} params.fields
```
Comment (aab-odoo) [inline addons/web/static/src/views/relational_model/static_list.js:236]: activeFields

File: addons/web/static/src/views/relational_model/static_list.js [hunk 3]
```diff
@@ -210,13 +216,130 @@ export class StaticList extends DataPoint {
...
+     * @param {Object} [params.context]
+     * @param {boolean} [params.withoutParent]
```
Comment (aab-odoo) [inline addons/web/static/src/views/relational_model/static_list.js:238]: record

File: addons/web/static/tests/views/fields/one2many_field_tests.js [hunk 4]
```diff
@@ -13065,4 +13065,71 @@ QUnit.module("Fields", (hooks) => {
         assert.containsOnce(target, ".o_list_view");
         assert.verifySteps([]);
     });
+
+    QUnit.tttt("abcde", async function (assert) {
```
Comment (pro-odoo) [inline addons/web/static/tests/views/fields/one2many_field_tests.js:13069]: 👀

**Review by pro-odoo — APPROVED** [review]

**Pull request merged**